# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/browser/nsWebBrowser.cpp
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc toolkit/components/browser/nsWebBrowser.cpp
--- a/toolkit/components/browser/nsWebBrowser.cpp	Wed Apr 28 11:09:38 2021 +0000
+++ b/toolkit/components/browser/nsWebBrowser.cpp	Wed Apr 28 11:26:49 2021 +0000
@@ -456,17 +456,17 @@
 }
 
 NS_IMETHODIMP
-nsWebBrowser::GoBack(bool aRequireUserInteraction) {
+nsWebBrowser::GoBack(bool aRequireUserInteraction, bool aUserActivation) {
   NS_ENSURE_STATE(mDocShell);
 
-  return mDocShell->GoBack(aRequireUserInteraction);
+  return mDocShell->GoBack(aRequireUserInteraction, aUserActivation);
 }
 
 NS_IMETHODIMP
-nsWebBrowser::GoForward(bool aRequireUserInteraction) {
+nsWebBrowser::GoForward(bool aRequireUserInteraction, bool aUserActivation) {
   NS_ENSURE_STATE(mDocShell);
 
-  return mDocShell->GoForward(aRequireUserInteraction);
+  return mDocShell->GoForward(aRequireUserInteraction, aUserActivation);
 }
 
 nsresult nsWebBrowser::LoadURI(const nsAString& aURI,
@@ -508,10 +508,10 @@
 }
 
 NS_IMETHODIMP
-nsWebBrowser::GotoIndex(int32_t aIndex) {
+nsWebBrowser::GotoIndex(int32_t aIndex, bool aUserActivation) {
   NS_ENSURE_STATE(mDocShell);
 
-  return mDocShell->GotoIndex(aIndex);
+  return mDocShell->GotoIndex(aIndex, aUserActivation);
 }
 
 NS_IMETHODIMP